backend/alerts/alerts.go

```diff
@@ -39,25 +41,38 @@ func SendErrorAlert(event SendErrorAlertEvent) error {
 		VisitedURL:      event.VisitedURL,
 	}
 
-	if !isWorkspaceIntegratedWithDiscord(*event.Workspace) {
+	var g errgroup.Group
+	g.Go(func() error {
+		for _, wh := range event.ErrorAlert.WebhookDestinations {
```

Comment:
Looks like you have to do the same thing as I did with discord and have to dupe all this looping. I thought about having some type of entry point like

```go
func SendAlert(webhookDestination event.ErrorAlert.WebhookDestinations, payload interface{}, alertType AlertType)
```

and then ideally, it would delegate to the correct function given the `alertType`. But I couldn't figure out a way to dynamically call a function in Go. 

Not sure if you know how to pull that off or a simpler way but it would be nice to tackle this as we continue to add more integrations.

Reply:
good point eric, on the webhook package side, it's just a matter of abstracting the json marshalling out to dedup all the `Send...` functions. let me try that

Reply:
looked into this a bit - we could dynamically call functions by having them be in a `map[string]func()` but since we have different payload types here that becomes more complex with getting the types to match up. going to leave this for now as it would require a lot of refactoring here to the payloads (rewriting the payloads to be an interface implementing a `Send()` method that could be called, for instance)